pilot/pkg/serviceregistry/serviceentry/controller.go

```diff
@@ -904,10 +904,11 @@ func autoAllocateIPs(services []*model.Service) []*model.Service {
 		// 3. the hostname is not a wildcard
 		if svc.DefaultAddress == constants.UnspecifiedIP && !svc.Hostname.IsWildCarded() &&
 			svc.Resolution != model.Passthrough {
-			hash.Write([]byte(makeServiceKey(svc)))
+			h.Reset()
+			h.Write([]byte(makeServiceKey(svc)))
 			// First hash is calculated by
-			s := hash.Sum32()
-			firstHash := s % uint32(maxIPs)
+			s := h.Sum64()
```

Comment:
I can not remember, but I ran in to some issue with Sum64 because of size.

Reply:
"github.com/cespare/xxhash/v2" does not support uint32 at all.

Reply:
It is better if you can recall what's the problem. IMO. uint64 and uint32 has no much difference in this case, we will mod maxIps(255 * 255) here. the conflict probability is similar in theory.

Reply:
I know what the problem, it seems we can never cal all values covering [0, 255*255)

Reply:
will look into how we can change the secondary hash

Reply:
correct. That is why I used 32.

Reply:
Tested manually, sum64 is slower than sum32.